Establish consistent label behavior across all parameters

erica opened the pull request:
We propose to normalize the first parameter declaration in methods
and functions. In this proposal, first parameter declarations will match
the existing behavior of the second and later parameters.
All parameters, regardless of position, will behave
uniformly. This will create a simple, consistent approach to parameter
declaration throughout the Swift programming language and bring
method and function declarations in-sync with initalizers, which
already use this standard.

e78l commented:
@erica Should the end of line 31 read `foo(a:b:c:)`?
declares `foo(_:b:c:)` and not `foo(a:b:c)`.

erica commented:
It is definitely missing a colon.

e78l commented:
In case there're any future changes :-)

erica commented:
I need to learn how to merge commit --append and then force something push something blah blah. I'll get it wrong

-- E


> On Mar 9, 2016, at 7:36 PM, Will Stanton <notifications@github.com> wrote:
> 
> In case there're any future changes :-)
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/apple/swift-evolution/pull/200#issuecomment-194628380>.
>

lattner commented:
I merged this as SE-0046, added the colon, and dropped two authors to simplify it.

erica commented:
https://github.com/apple/swift-evolution/blob/master/proposals/0046-first-label.md <https://github.com/apple/swift-evolution/blob/master/proposals/0046-first-label.md> looks good.

-- E

> On Mar 9, 2016, at 8:53 PM, Chris Lattner <notifications@github.com> wrote:
> 
> I merged this as SE-0046, added the colon, and dropped two authors to simplify it.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/apple/swift-evolution/pull/200#issuecomment-194650964>.
>

lattner commented:
Sorry, fighting git tonight.  Try now.
